### PR TITLE
Use sylabs/sif new implementation for all SIF access in this tree.

### PR DESCRIPTION
**Pull request by yhcote**
Remove old references to sif C and use sylabs/sif to create and load containers.

This patch enabled the removal of a lot of unused C files.

### CODE REVIEW

File: src/pkg/build/assembler_sif.go [hunk 1]
```diff
@@ -37,17 +93,14 @@ func (a *SIFAssembler) Assemble(b *Bundle, path string) (err error) {
 	mksquashfsCmd.Stderr = os.Stderr
 	err = mksquashfsCmd.Run()
 	if err != nil {
-		return err
+		return
 	}
+	defer os.Remove(squashfsPath)
```
Comment (cclerget) [inline src/pkg/build/assembler_sif.go]: I think defer should be set before error check here to be sure squashfs temporary file is deleted, sif command could failed for some reasons and have already created temporary file

File: src/pkg/build/assembler_sif.go [hunk 2]
```diff
@@ -6,23 +6,79 @@
...
+	return nil
+}
+
 // Assemble creates a SIF image from a Bundle
 func (a *SIFAssembler) Assemble(b *Bundle, path string) (err error) {
 	mksquashfs, err := exec.LookPath("mksquashfs")
 	if err != nil {
 		sylog.Errorf("mksquashfs is not installed on this system")
-		return err
+		return
```
Comment (cclerget) [inline src/pkg/build/assembler_sif.go:81]: Is it intentional to not return err ?

File: src/pkg/build/assembler_sif.go [hunk 3]
```diff
@@ -37,17 +93,14 @@ func (a *SIFAssembler) Assemble(b *Bundle, path string) (err error) {
 	mksquashfsCmd.Stderr = os.Stderr
 	err = mksquashfsCmd.Run()
 	if err != nil {
-		return err
+		return
```
Comment (cclerget) [inline src/pkg/build/assembler_sif.go]: And here too ?

File: src/pkg/build/assembler_sif.go [hunk 4]
```diff
@@ -37,17 +93,14 @@ func (a *SIFAssembler) Assemble(b *Bundle, path string) (err error) {
...
 
-	sifCmd := exec.Command("singularity", "sif", "create", "-P", squashfsPath, "-f", "SQUASHFS", "-p", "SYSTEM", "-c", "LINUX", path)
-	sifCmd.Stdin = os.Stdin
-	sifCmd.Stdout = os.Stdout
-	sifCmd.Stderr = os.Stderr
-	err = sifCmd.Run()
+	err = createSIFSinglePart(path, squashfsPath)
 	if err != nil {
-		return err
+		return
```
Comment (cclerget) [inline src/pkg/build/assembler_sif.go:102]: same here ?

File: src/pkg/image/image.go [hunk 5]
```diff
@@ -112,7 +112,9 @@ func Init(path string, writable bool) (*Image, error) {
 	if writable {
 		flags = os.O_RDWR
 	}
-	resolvedPath, err := filepath.EvalSymlinks(path)
+
+	abspath, _ := filepath.Abs(path)
```
Comment (cclerget) [inline src/pkg/image/image.go]: My bad ! I forgot but it may be useful to check error and return it

File: src/pkg/build/assembler_sif.go [hunk 6]
```diff
@@ -6,23 +6,79 @@
...
+	}
+
+	return nil
+}
+
 // Assemble creates a SIF image from a Bundle
 func (a *SIFAssembler) Assemble(b *Bundle, path string) (err error) {
 	mksquashfs, err := exec.LookPath("mksquashfs")
 	if err != nil {
 		sylog.Errorf("mksquashfs is not installed on this system")
```
Comment (cclerget) [inline src/pkg/build/assembler_sif.go:80]: why not have something like `return fmt.Errorf("mksquashfs is not installed on this system")` rather and let calling function display error ?

**Review by cclerget — CHANGES_REQUESTED** [review]
Have some requests mainly for assembler_sif, otherwise looks good to me

File: src/pkg/build/assembler_sif.go (hunk 2 above)
Comment (yhcote) [reply src/pkg/build/assembler_sif.go:81]: it's a naked return. This is done when the return type is named. e.g., (err error)

File: src/pkg/build/assembler_sif.go (hunk 3 above)
Comment (yhcote) [reply src/pkg/build/assembler_sif.go]: same comment ;)

File: src/pkg/build/assembler_sif.go (hunk 4 above)
Comment (yhcote) [reply src/pkg/build/assembler_sif.go:102]: And same here too yes :)

File: src/pkg/build/assembler_sif.go (hunk 6 above)
Comment (yhcote) [reply src/pkg/build/assembler_sif.go:80]: good point but it's not part of my patch ? ;)

File: src/pkg/build/assembler_sif.go (hunk 2 above)
Comment (cclerget) [reply src/pkg/build/assembler_sif.go:81]: Oh I missed that ! Sorry !

File: src/pkg/build/assembler_sif.go (hunk 6 above)
Comment (cclerget) [reply src/pkg/build/assembler_sif.go:80]: Yes right ! :D

File: src/pkg/image/image.go (hunk 5 above)
Comment (yhcote) [reply src/pkg/image/image.go]: ok no problem

File: src/pkg/build/assembler_sif.go (hunk 1 above)
Comment (yhcote) [reply src/pkg/build/assembler_sif.go]: true.

**Review by cclerget — APPROVED** [review]

**Pull request merged**